phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/managers/PeerManager.kt

```diff
@@ -42,6 +51,13 @@ class PeerManager(
     private val _balance = MutableStateFlow<MilliSatoshi?>(null)
     val balance: StateFlow<MilliSatoshi?> = _balance
 
+    private val _swapInWallet = MutableStateFlow<WalletState?>(null)
+    private val _pendingReservedUtxos = MutableStateFlow<Map<ByteVector32, List<Utxo>>>(emptyMap())
+    private val _reservedUtxos = MutableStateFlow<Set<Utxo>>(emptySet())
+
+    private val _swapInWalletBalance = MutableStateFlow(WalletBalance.empty())
+    val swapInWalletBalance: StateFlow<WalletBalance> = _swapInWalletBalance
```

Comment:
`PaymentsManager` also listens to `nodeEvents` to maintain a map of pending swap-ins, which is conceptually similar to this swap-in balance. It could be moved here. Or even better, what do you think of creating a new `BalanceManager` class that would be a single point of entry for everything balance related?

Reply:
> what do you think of creating a new `BalanceManager` class 

Love it. Done it [477cf84](https://github.com/ACINQ/phoenix/pull/316/commits/477cf845d7bd89f279b15ffddf43e0ae40eb1fb8)